src/utils/components/SelectSnapshot/utils.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { ReactElement } from 'react';
+
+import { SelectOption } from '@patternfly/react-core';
+
+export const FilterSelect = (options: string[]) => {
+  return (_, value: string): ReactElement[] => {
+    let newOptions = options;
+
+    if (value) {
+      const regex = new RegExp(value, 'i');
+      newOptions = options.filter((namespace) => regex.test(namespace));
+    }
+
+    return newOptions.map((namespace) => (
+      <SelectOption key={namespace} value={namespace} />
+    )) as ReactElement[];
```

Comment:
hmm returning elements from util function isnt recommended, what about return the array and in the consuming component just map over it? like we are doing with dropdown items or other select items?